### PR TITLE
Update prettier to v2

### PR DESCRIPTION
**Pull request by zxbodya**
This is needed in context of migration to typescript #11578 to allow using `import type` typescript syntax.

| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | 
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  | Yes, updating prettier
| License                  | MIT

### CODE REVIEW

**Comment by existentialism** [comment]
FWIW, we've held off on this for the same reasons as jest@25 (https://github.com/babel/babel/pull/11115).

**Comment by zxbodya** [comment]
> FWIW, we've held off on this for the same reasons as jest@25 (#11115).

should be a bit better, as I understand it is used only during lint and when generating some code during build…

so, at least at the first sight, it should not be needed when running the test (as an option might be good enough to add polyfill during the build, or skip regenerating the files(which are committed anyway)

File: scripts/utils/formatCode.js [hunk 1]
```diff
@@ -1,11 +1,22 @@
...
-  const prettierConfig = prettier.resolveConfig.sync(filename);
-  prettierConfig.filepath = filename;
-  prettierConfig.parser = "babel";
+// TODO: Remove this `if` in Babel 8
+// Prettier only supports Node.js 10+, so we can fallback to not formatting
+// o CI on older Node.js versions
 
-  return prettier.format(code, prettierConfig);
-};
+if (process.env.CI && parseInt(process.versions.node, 10) < 10) {
```
Comment (nicolo-ribaudo) [inline scripts/utils/formatCode.js:7]: We use prettier format the generated type definition files. However, we don't really need to do this on CI and we can skip it on Node.js 6 and 8 without introducing problems.

Comment (zxbodya) [reply scripts/utils/formatCode.js:7]: looks good for me, was thinking about disabling code generation completely - but, I think, having it without prettier is a better option

**Comment by nicolo-ribaudo** [comment]
For reviewers: 19a7acf4bd6351574f61d93b72653de71f2771a2 is what makes CI work on Node.js 6

**Comment by nicolo-ribaudo** [comment]
Let's wait for Brian's go-ahead to check if we can merge this (https://github.com/babel/babel/pull/11579#issuecomment-630190639)

**Comment by existentialism** [comment]
:shipit: !

**Comment by nicolo-ribaudo** [comment]
Thanks @zxbodya!

**Pull request merged**